biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java

```diff
@@ -338,17 +339,25 @@ private boolean isMulti(String name) {
 	}
 
 	private Map<Archive, Resource> parseMulti(Archive archive, File multi) throws Exception {
+		Set<String> multis = new HashSet<String>();
+		File localArchives = repo.toLocalFile(archive)
+			.getParentFile();
 		Map<Archive, Resource> result = new HashMap<>();
 		try (Jar jar = new Jar(multi)) {
-			int n = 1000;
 			for (Entry<String, aQute.bnd.osgi.Resource> entry : jar.getResources()
 				.entrySet()) {
 				String path = entry.getKey()
 					.toLowerCase();
 				if (path.endsWith(".jar")) {
+					String name = Strings.lastPathSegment(path)[1];
+					name = Strings.extension(name)[0];
+					name = ZipUtil.cleanPath(name);
+					name = name.replaceAll("[.,;\"'-]+", "_");
+					while (!multis.add(name)) {
+						name += "_";
```

Comment:
```suggestion
						name = name.concat("_");
```

Reply:
done

Reply:
This block is a very rare occurrence and it is less readable. I am not in favor in making code less readable for an immeasurable performance gains? If performance is that important, should a string builder then not even be better?